src/data_gradients/feature_extractors/utils.py

```diff
@@ -1,40 +1,67 @@
 
-        value_type = type(train_value) if train_value is not None else type(val_value)
-        default_value = value_type()
+        :param df:          The DataFrame to get the top values from.
+        :param id_col:      The name of the id column.
+        :param split_col:   The name of the split column. (Usually 'split')
+        :param value_col:   The name of column that will be used to calculate the metric.
+        :return: The Dataframe with only the rows associated to the most important values.
+        """
+        # Verify inputs
+        for col in [id_col, split_col, value_col]:
+            if col not in df.columns:
+                raise ValueError(f"{col} is not a column in the DataFrame")
 
-        aligned_train_histogram[key] = train_value or default_value
-        aligned_val_histogram[key] = val_value or default_value
+        # the mean of val_col for each id_col/split_col
+        df_mean = df.groupby([id_col, split_col])[value_col].mean().reset_index()
 
-    return aligned_train_histogram, aligned_val_histogram
+        # Pivot DataFrame to have 'train' and 'val' as columns
+        df_pivot = df_mean.pivot(index=id_col, columns=split_col, values=value_col)
 
+        # Calculate the relative difference or average based on the prioritization_mode
+        if self.prioritization_mode == "train_val_diff":
+            df_pivot["metric"] = np.abs((df_pivot["train"] - df_pivot["val"]) / ((df_pivot["train"] + df_pivot["val"]) / 2))
+        elif self.prioritization_mode in ["outliers", "max", "min", "min_max"]:
+            df_pivot["metric"] = (df_pivot["train"] + df_pivot["val"]) / 2
 
-def normalize_values_to_percentages(counters: List[float], total_count: float) -> List[float]:
-    """
-    Normalize a list of count to percentages relative to a total value.
+        if self.prioritization_mode == "outliers":
+            mean, std = df_pivot["metric"].mean(), df_pivot["metric"].std()
+            df_pivot["metric"] = (df_pivot["metric"] - mean).abs() / std
 
-    :param counters:    Values to normalize.
-    :param total_count: Total number of values, which will be used to calculate percentages.
-    :return:            Values representing the percentages of each input value.
-    """
-    if total_count == 0:
-        total_count = 1
-    return [np.round(((100 * count) / total_count), 3) for count in counters]
+        # Only return the top k.
+        if self.prioritization_mode in ["train_val_diff", "outliers", "max"]:
+            top_ids = df_pivot.nlargest(self.topk, "metric").index
+            return df[df[id_col].isin(top_ids)]
+        elif self.prioritization_mode == "min":
+            top_ids = df_pivot.nsmallest(self.topk, "metric").index
+            return df[df[id_col].isin(top_ids)]
+        elif self.prioritization_mode == "min_max":
+            n_min_results = self.topk // 2
+            bottom_ids = df_pivot.nlargest(n_min_results, "metric").index
```

Comment:
What if len(df_pivot) is < top_k?

Reply:
Fixed it here [01bde99](https://github.com/Deci-AI/data-gradients/pull/129/commits/01bde99fa4a099d88bb2c31c142532a183fb3fcb)